front/app/components/EsriMap/utils.tsx

```diff
@@ -44,10 +44,24 @@ export const getDefaultBasemap = (tileProvider: string | undefined): Layer => {
       },
     });
   }
-  return new WebTileLayer({
+  const webTileLayer = new WebTileLayer({
     urlTemplate: tileProvider || DEFAULT_TILE_PROVIDER,
     copyright: getTileAttribution(tileProvider || ''),
   });
+
+  if (tileProvider?.includes('maptiler')) {
+    webTileLayer.set('tileInfo.size', 512);
+    webTileLayer.set(
+      'tileInfo.lods',
+      webTileLayer.tileInfo.lods.map((lod) => {
+        lod.resolution = lod.resolution / 2;
+        lod.scale = lod.scale / 2;
+        return lod;
+      })
+    );
+  }
+
+  return webTileLayer;
```

Comment:
The problem was that we were fetching the tiles at a size of 512px but treating them as 256px in the Esri map. I've now explicitly set the size and Level of Detail (lods) to match 512opx which fixes the problem.